File: Getting Started - WCF RIA Services/[VB]-Getting Started - WCF RIA Services/VB/HRApp/Models/UserExtensions.vb
Imports System
Imports System.ComponentModel

Namespace Web

    Partial Public Class User
#Region "Make DisplayName Bindable"
        Protected Overrides Sub OnPropertyChanged(ByVal e As System.ComponentModel.PropertyChangedEventArgs)
            If e Is Nothing Then
                Throw New ArgumentNullException("e")
            End If

            MyBase.OnPropertyChanged(e)

            If e.PropertyName = "Name" Or e.PropertyName = "FriendlyName" Then
                Me.RaisePropertyChanged("DisplayName")
            End If
        End Sub
#End Region
    End Class
End Namespace

File: Getting Started - WCF RIA Services/[C#]-Getting Started - WCF RIA Services/C#/HRApp/Models/UserExtensions.cs
namespace HRApp.Web
{
    using System;
    using System.ComponentModel;

    public partial class User
    {
        #region Make DisplayName Bindable
        protected override void OnPropertyChanged(PropertyChangedEventArgs e)
        {
            if (e == null)
            {
                throw new ArgumentNullException("e");
            }

            base.OnPropertyChanged(e);

            if (e.PropertyName == "Name" || e.PropertyName == "FriendlyName")
            {
                this.RaisePropertyChanged("DisplayName");
            }
        }
        #endregion
    }
}
